### PR TITLE
Bug 1822193 - Scroll to the active tab when opening the Tabs Tray

### PR DESCRIPTION
**Pull request by MozillaNoah**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1822193

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt [hunk 1]
```diff
@@ -145,60 +146,77 @@ fun TabsTray(
...
+                                onTabClick = handleTabClick,
+                                onTabLongClick = onTabLongClick,
+                                header = optionalInactiveTabsHeader,
+                            )
+                        } else {
+                            Text(
+                                text = "Empty state",
+                                color = FirefoxTheme.colors.textPrimary,
+                                style = FirefoxTheme.typography.body1,
+                            )
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt:198]: I had to add a placeholder empty state UI before the official ticket because of the `TabLayout` getting composed when there's no tabs, breaking the initial scroll because of how the tab lists are populated in `TabsTrayStore`

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayTabLayouts.kt [hunk 2]
```diff
@@ -59,10 +59,20 @@ fun TabLayout(
...
     header: (@Composable () -> Unit)? = null,
 ) {
+    var selectedTabIndex = 0
+    selectedTabId?.let {
+        tabs.forEachIndexed { index, tab ->
+            if (tab.id == selectedTabId) {
+                selectedTabIndex = index
+            }
+        }
+    }
```
Comment (MatthewTighe) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayTabLayouts.kt:69]: nit: We could avoid a `var` here with something like:
```
val selectedTabIndex = tabs.indexOf(tabs.find { tab ->
    tab.id == selectedTabId
})
```

Feels a little hacky with all the nullability implied in there, but might be preferable.

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt (hunk 1 above)
Comment (MatthewTighe) [reply fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt:198]: This could be done in the empty state ticket, but I think we should split these conditional cases into their own functions for readability. Something like:

```
if (normalTabs.isNotEmpty() || inactiveTabs.isNotEmpty()) {
    TabsUI()
} else {
    EmptyUI()
}
```

That, or the empty UI case should come first. My complaint is mostly that the nesting in the `TabsLayout` case makes it a little hard to parse

Comment (MatthewTighe) [reply fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt:198]: I do also see the "Empty state" text flash briefly when loading the tray, but I see that that is not technically a functionality degradation vs the XML version. Still, I can't help wondering if there's a way we could distinguish a "loading" state from an "empty" state to avoid that flash of text.

**Review by MatthewTighe — APPROVED** [review]
Couple nits, but this otherwise LGTM.

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt (hunk 1 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt:198]: For sure. We almost kinda need a tri-state here of:
1. Empty
2. Loading
3. Populated

I'm not immediately sure how to engineer that with the store observations.

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayTabLayouts.kt (hunk 2 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayTabLayouts.kt:69]: I wanted to avoid this, since doing `indexOf` + `tabs.find` causes two linear iterations over the list of tabs, but also `find` can return `-1` when the tab isn't in the list (thinking of the situation it's searching for a Private tab in the normal list) which would cause an exception.
